frontend/src/routes/train.jsx

```diff
@@ -0,0 +1,120 @@
+import { styled } from "solid-styled-components";
+import { createFormGroup, createFormControl } from "solid-forms";
+import Button from "../components/form/button";
+import EmailInput from "../components/form/email-input";
+import HashInput from "../components/form/hash-input";
+import FileInput from "../components/form/file-input";
+import Slider from "../components/slider";
+import { getFilesUrls, validateEmail } from "../helpers";
+import { postTrain } from "../api";
+import { config } from "../config";
+
+const Train = () => {
+  const group = createFormGroup({
+    email: createFormControl("", {
+      validators: (value) =>
+        validateEmail(value)
+          ? null
+          : { errorMessage: "Email address is invalid" },
+    }),
+    files: createFormControl([], {
+      validators: (files) =>
+        files.length < config.minFiles
```

Comment:
Please check file extensions too as we want files to be images. Allowed formats are JPEG and PNG for now.